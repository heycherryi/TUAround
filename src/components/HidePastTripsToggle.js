import React from 'react';
import PropTypes from 'prop-types';

function HidePastTripsToggle({ hidePastTrips, onChange }) {
  return (
    <div className="my-2">
      <div className="form-check form-switch">
        <input className="form-check-input" type="checkbox" id="hidePastTrips" checked={hidePastTrips} onChange={onChange} />
        <label className="form-check-label" htmlFor="hidePastTrips">Hide past trips</label>
      </div>
    </div>
  );
}

HidePastTripsToggle.propTypes = {
  hidePastTrips: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

HidePastTripsToggle.defaults = {
  hidePastTrips: true,
};

export default HidePastTripsToggle;
