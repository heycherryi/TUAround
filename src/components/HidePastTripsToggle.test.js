/* globals test, expect */

import React from 'react';
import { render, screen } from '@testing-library/react';
import HidePastTripsToggle from './HidePastTripsToggle';

test('renders HidePastTripsToggle', () => {
  const { container } = render(
    <HidePastTripsToggle hidePastTrips={true} onChange={() => console.log('Changed!')} />,
  );
  expect(screen.getByRole('checkbox')).toBeChecked();
  expect(screen.getByText('Hide past trips')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('renders HidePastTripsToggle unchecked', () => {
  const { container } = render(
    <HidePastTripsToggle hidePastTrips={false} onChange={() => console.log('Changed!')} />,
  );
  expect(screen.getByRole('checkbox')).not.toBeChecked();
  expect(screen.getByText('Hide past trips')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
