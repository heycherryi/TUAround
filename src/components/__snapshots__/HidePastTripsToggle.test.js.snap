// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HidePastTripsToggle 1`] = `
<div>
  <div
    class="my-2"
  >
    <div
      class="form-check form-switch"
    >
      <input
        checked=""
        class="form-check-input"
        id="hidePastTrips"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="hidePastTrips"
      >
        Hide past trips
      </label>
    </div>
  </div>
</div>
`;

exports[`renders HidePastTripsToggle unchecked 1`] = `
<div>
  <div
    class="my-2"
  >
    <div
      class="form-check form-switch"
    >
      <input
        class="form-check-input"
        id="hidePastTrips"
        type="checkbox"
      />
      <label
        class="form-check-label"
        for="hidePastTrips"
      >
        Hide past trips
      </label>
    </div>
  </div>
</div>
`;
